src/util/io.c

```diff
@@ -59,6 +59,10 @@ read_all (
                          fd, errno, strerror (errno));
             return recvd_total;
         }
+        if (recvd == 0) {
+            LOG_WARNING ("read on fd %d returned 0 / EOF", fd);
+            return recvd_total;
```

Comment:
Here I would also return `recvd` for consistency, even that for EOF will be semantically the same than `recvd_total`.

Reply:
I've added a comment to the function to clarify this. For now I'm just trying to keep the behavior consistent. Happy to revisit the behavior of this function though. I figured this would be the case anyways when async I/O gets implemented.